### PR TITLE
Create journal-of-porphyrins-and-phthalocyanines.csl

### PR DESCRIPTION
**Pull request by jakuhne**
Submitting new style by personal request and action

### CODE REVIEW

File: journal-of-porphyrins-and-phthalocyanines.csl [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+      </if>
+    </choose>
+  </macro>
+  <macro name="title">
+    <choose>
+      <if type="bill book graphic legal_case legislation motion_picture report song" match="any">
+        <text variable="title" text-case="title"/>
+      </if>
+      <else>
+        <text value=""/>
```
Comment (adam3smith) [inline journal-of-porphyrins-and-phthalocyanines.csl]: What's the goal here? If you actually don't want a title for some item types, you can just not have an `else` condition.

File: journal-of-porphyrins-and-phthalocyanines.csl [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+  <macro name="editor">
+    <names variable="editor">
+      <name delimiter-precedes-et-al="never" delimiter-precedes-last="never" initialize-with="" name-as-sort-order="all" sort-separator=" "/>
+      <label form="short" prefix=", "/>
+    </names>
+  </macro>
+  <macro name="author">
+    <group suffix=".">
+      <names variable="author">
+        <name and="text" delimiter-precedes-et-al="never" delimiter-precedes-last="never" et-al-min="19" et-al-use-first="18" et-al-use-last="true" initialize-with="" name-as-sort-order="all" sort-separator=" "/>
```
Comment (adam3smith) [inline journal-of-porphyrins-and-phthalocyanines.csl]: What are the et al rules here? You have 19/18 here and then 4/3 below and you also use the `et-al-use-last`, which would be unusual for chemistry.

File: journal-of-porphyrins-and-phthalocyanines.csl (hunk 1 above)
Comment (jakuhne) [reply journal-of-porphyrins-and-phthalocyanines.csl]: Thank you for pointing these out. If you can't tell I'm new to this. I only have a little experience and that's with java. I started on this before I realized that I could ask someone else to make these. I used the visual editor so I was uncertain about just deleting functions.

Comment (adam3smith) [reply journal-of-porphyrins-and-phthalocyanines.csl]: I removed this already in the style, so only thing left is the et-al quesiton.

File: journal-of-porphyrins-and-phthalocyanines.csl (hunk 2 above)
Comment (jakuhne) [reply journal-of-porphyrins-and-phthalocyanines.csl]: The 18/19 is the limit on Mendeley's visual editor.  I definitely didn't want a limit and I think it got cranky when I tried turning the et-al-use-last off. Something like that. Thank you.

Comment (adam3smith) [reply journal-of-porphyrins-and-phthalocyanines.csl]: So just to clarify, the style should never use et-al in the bibliography?

Comment (jakuhne) [reply journal-of-porphyrins-and-phthalocyanines.csl]: The guidelines don't explicitly say so but looking at a published review it looks like et-al is never used. 

"Journals should be cited in the following way: author surnames and initials (no dot after initial), insert “and” between the last two authors (with no comma), dot, journal name (italicized and abbreviated in accordance with the ACS Style Guide..."

**Comment by jakuhne** [comment]
Thank you for cleaning up my crappy code @adam3smith!

**Comment by adam3smith** [comment]
Thanks -- this actually looked very good from the start, nothing crappy about it.

**Comment by POBrien333** [comment]
@jakuhne

Well done! And you got lucky your style god merged so quickly! (I was helping Jeremy via email as he contacted as he based the style on one I made. ) :)

**Pull request merged**